lib/brakeman/parsers/rails3_erubis.rb

```diff
@@ -9,23 +9,14 @@ def add_preamble(src)
 
   #This is different from Rails 3 - fixes some line number issues
   def add_text(src, text)
-    if text =~ /\A\n+\z/
-      src << text
-    elsif text.include? "\n"
-      lines = text.split("\n")
-      if text.match(/\n\z/)
-        lines.each do |line|
-          src << "@output_buffer << ('" << escape_text(line) << "'.html_safe!);\n"
-        end
+    lines = text.lines
```

Comment:
Unlike "text.split("\n")`, the `String#lines` method will split things in a predictable way across newlines.